dolphin/drivers/dell_emc/vmax/client.py

```diff
@@ -123,3 +123,58 @@ def list_pools(self):
             LOG.error("Failed to get pool metrics from VMAX: {}".format(err))
             raise exception.StorageBackendException(
                 reason='Failed to get pool metrics from VMAX')
+
+    def list_volumes(self, storage_id):
+
+        try:
+            # List all volumes
+            volumes = self.conn.provisioning.get_volume_list(filters={'data_volume': 'false'})
+
+            volume_list = []
+            for volume in volumes:
+                # Get volume details
+                vol = self.conn.provisioning.get_volume(volume)
+
+                total_cap = vol['cap_mb'] * units.Mi
+                used_cap = (total_cap * vol['allocated_percent']) / 100.0
+                free_cap = total_cap - used_cap
+
+                # TODO: Update constants.VolumeStatus to make mapping more precise
+                switcher = {
```

Comment:
Please move it to Constants file

Reply:
This is currently used only here, also I think, it is better readable this way.

Reply:
Ok